crates/turbo-tasks/src/manager.rs

```diff
@@ -400,18 +397,25 @@ impl<B: Backend> TurboTasks<B> {
                     if this.stopped.load(Ordering::Acquire) {
                         return false;
                     }
-                    if let Some(execution) = this.backend.try_start_task_execution(task_id, &*this)
-                    {
-                        // Setup thread locals
-                        let (result, duration, instant) = CELL_COUNTERS
-                            .scope(Default::default(), async {
-                                let (result, duration, instant) = TimedFuture::new(
-                                    AssertUnwindSafe(execution.future).catch_unwind(),
+
+                    // Setup thread locals
+                    if let Some((result, duration, instant)) = CELL_COUNTERS
+                        .scope(Default::default(), async {
+                            if let Some(execution) =
+                                this.backend.try_start_task_execution(task_id, &*this)
+                            {
+                                Some(
+                                    TimedFuture::new(
+                                        AssertUnwindSafe(execution.future).catch_unwind(),
+                                    )
+                                    .await,
                                 )
-                                .await;
-                                (result, duration, instant)
-                            })
-                            .await;
+                            } else {
+                                None
+                            }
+                        })
+                        .await
```

Comment:
This condition is hard to read. Extract the right-hand side into its own variable.

e.g.

```rust
let execution_future = CELL_COUNTERS.scope(...);
if let Some((result, duration, instant)) = execution_future.await {
```